src/shims/unix/linux/sync.rs

```diff
@@ -153,13 +153,12 @@ pub fn futex<'tcx>(
                     Scalar::from_target_isize(0, this), // retval_succ
                     Scalar::from_target_isize(-1, this), // retval_timeout
                     dest.clone(),
-                    this.eval_libc("ETIMEDOUT"),
+                    LibcError("ETIMEDOUT"),
```

Comment:
This will require `futex_wait` to be changed: the type of `errno_timeout` should become `IoError`.